subprojects/build-operations/src/main/java/org/gradle/internal/operations/BuildOperationExecutionListener.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.operations;
+
+import javax.annotation.Nullable;
+
+interface BuildOperationExecutionListener {
+    void start(BuildOperationDescriptor descriptor, BuildOperationState operationState);
+
+    void stop(BuildOperationDescriptor descriptor, BuildOperationState operationState, @Nullable BuildOperationState parent, DefaultBuildOperationContext context);
```

Comment:
❓ Why is `DefaultBuildOperationContext` passed into this method and not `BuildOperationContext`?

Reply:
Also: How das the context relate to the state?

Reply:
I changed this around a bit and now we pass in an interface. We can't use `BuildOperationContext` because that's intentionally write-only.